backend/src/dependencies.rs

```diff
@@ -660,21 +665,19 @@ pub async fn add_dependent_to_current_dependents_lists<
     db: &mut Db,
     dependent_id: &PackageId,
     current_dependencies: I,
+    locks: &dyn CurrentDependentsLock,
 ) -> Result<(), Error> {
     for (dependency, dep_info) in current_dependencies {
-        if let Some(dependency_model) = crate::db::DatabaseModel::new()
-            .package_data()
-            .idx_model(&dependency)
-            .and_then(|pkg| pkg.installed())
-            .check(db)
+        let mut dependency_dependents = locks
```

Comment:
Removed the mutation model for now, I like this more but the real reason was for simplicity and POC